pkg/e2e/verify/user_webhook.go

```diff
@@ -27,15 +27,390 @@ var _ = ginkgo.Describe(userWebhookTestName, func() {
+				Groups: []string{
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, groups, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).NotTo(HaveOccurred())
+			err = deleteUser(user.Name, h)
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		ginkgo.It("system:admin can manage customer users with SRE IDP and no group", func() {
+			userName := util.RandomStr(5) + "@customdomain"
+			identities := []string{"OpenShift_SRE:" + util.RandomStr(5)}
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "system:admin",
+				Groups: []string{
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, []string{}, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).NotTo(HaveOccurred())
+			err = deleteUser(user.Name, h)
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		ginkgo.It("system:admin can manage customer users with other IDP and RH group", func() {
+			userName := util.RandomStr(5) + "@customdomain"
+			identities := []string{"otherIDP:testing_string"}
+			groups := []string{"osd-devaccess"}
+			// we need to add the username to the group before we create the user,
+			// because the user object cannot be created until the username is in
+			// the group
+			addUserToGroup(userName, groups[0], h)
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "system:admin",
+				Groups: []string{
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, groups, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).NotTo(HaveOccurred())
+			err = deleteUser(user.Name, h)
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		ginkgo.It("system:admin can manage customer users with other IDP and no group", func() {
+			userName := util.RandomStr(5) + "@customdomain"
+			identities := []string{"otherIDP:testing_string"}
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "system:admin",
+				Groups: []string{
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, []string{}, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+				deleteUser(user.Name, h)
+			}()
+			Expect(err).NotTo(HaveOccurred())
+			err = deleteUser(user.Name, h)
+			Expect(err).NotTo(HaveOccurred())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// not testing osd-sre-admins because RBAC prevents them from managing
+		// users, even though webhook allows them to
```

Comment:
I know this is a webhook test, so maybe not here, but should we have a test for this case and the below `oauth-openshift` to ensure that the RBAC does that? Again, probably not this PR but if we're calling it out, we could add a reference to the test that proves that in the comments here if/when they exist.

Reply:
right. I'm not sure where such a test should go. I don't want to put it here, because we would have to say that we expected a failure even though the webhook would allow this behavior - which would mean that if we changed the RBAC to also allow this behavior, the webhook e2e tests would break due to a change in a completely different/irrelevant repository. @lisa may have more ideas on this?

Reply:
Let's take this into slack, and we can create a card if needed there.